### PR TITLE
Improve ECC PKCS#11 tests

### PR DESCRIPTION
**Pull request by boricm**
* try catch in p11 high and low level tests. If a test throws an exception, the test output contains the name of the test case that failed. Before, only the name of the suite was printed.
* add a test case that imports elliptic curve parameters
* add curve brainpool512r1 to PKCS11 test test_ecdsa_sign_verify()
* add curve brainpool512r1 to test_ecdsa_generate_keypair()
* update tested/supported smartcards in manual

### CODE REVIEW

File: src/tests/test_pkcs11_high_level.cpp [hunk 1]
```diff
@@ -93,14 +93,21 @@ class TestSession
 Test::Result test_module_ctor()
    {
    Test::Result result("Module ctor");
+   
+   try
```
Comment (randombit) [inline src/tests/test_pkcs11_high_level.cpp]: I understand the motivation for this but I think it is possible to do in a simpler way that involves less boilerplate. Change `run_pkcs11_tests` to take `std::vector<std::pair<std::string, std::function<Test::Result()>>>& fns` where fns[i].first is the name and fns[i].second in the function to invoke. Then the try/catch that is already in `run_pkcs11_tests` can include the function name as well as the test name in the event of any exceptions.

**Comment by randombit** [comment]
@boricm Friendly ping. No worries if you don't have time to make these adjustments right away but I wanted to check that they at least made sense. Thanks!

**Comment by boricm** [comment]
> @boricm Friendly ping. No worries if you don't have time to make these adjustments right away but I wanted to check that they at least made sense. Thanks!

@randombit Sorry, I'm a litte busy a the moment. I really appreciate your input, especially your suggestion to avoid the boilerplate! I am definitely going to fix it as soon as I can ;-)

File: src/tests/test_pkcs11_high_level.cpp [hunk 2]
```diff
@@ -93,14 +93,21 @@ class TestSession
...
+	   });
 
-   result.test_throws("Module ctor fails for non existent path", []()
-      {
-      Module failing_module("/a/b/c");
-      });
-
-   Module module(Test::pkcs11_lib());
-   result.test_success("Module ctor did not throw and completed successfully");
+	   Module module(Test::pkcs11_lib());
```
[the remark this answers is not in the document]
Comment (boricm) [reply src/tests/test_pkcs11_high_level.cpp]: fixed

File: src/tests/test_pkcs11_high_level.cpp [hunk 3]
```diff
@@ -93,14 +93,21 @@ class TestSession
 Test::Result test_module_ctor()
```
[the remark this answers is not in the document]
Comment (boricm) [reply src/tests/test_pkcs11_high_level.cpp]: fixed

File: src/tests/test_pkcs11_high_level.cpp (hunk 1 above)
Comment (boricm) [reply src/tests/test_pkcs11_high_level.cpp]: @randombit I implemented your suggestion, thanks for the hint! What do you think about it?
P.S.: I will be offline until Monday. Have a nice weekend! :-)

**Pull request merged**